pkg/controllers/gpu/controller.go

```diff
@@ -183,9 +225,52 @@ func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	if devicePlugin.Spec.InitImage != "" {
 		setInitContainer(&daemonSet.Spec.Template.Spec, devicePlugin.Spec.InitImage)
 	}
+	// add service account if resource manager is enabled
+	if devicePlugin.Spec.ResourceManager {
+		daemonSet.Spec.Template.Spec.ServiceAccountName = serviceAccountName
+		addVolumeIfMissing(&daemonSet.Spec.Template.Spec, "podresources", "/var/lib/kubelet/pod-resources", v1.HostPathDirectory)
+		addVolumeMountIfMissing(&daemonSet.Spec.Template.Spec, "podresources", "/var/lib/kubelet/pod-resources")
+	}
 	return &daemonSet
 }
 
+func addVolumeMountIfMissing(spec *v1.PodSpec, name, mountPath string) {
+	missing := true
+	for _, mount := range spec.Containers[0].VolumeMounts {
+		if mount.Name == name {
+			missing = false
+			break
+		}
```

Comment:
As this is now its own function, just return here and remove the "missing" variable.

Reply:
done